contrib/murmurhash/include/MurmurHash3.h

```diff
@@ -23,10 +24,6 @@ typedef unsigned __int64 uint64_t;
 
 #endif // !defined(_MSC_VER)
 
-#ifdef __cplusplus
```

Comment:
I could not understand how this was intended to work.
MurmurHash3.cpp uses C++ names mangling.

Reply:
See https://github.com/ClickHouse/ClickHouse/pull/8792

Maybe it's enough to specify `extern "C"` in declaration.

Reply:
I still do not understand.
When MurmurHash3 will be compiled, these functions will be mangled in C++-style, not in C.

Reply:
```
milovidov@milovidov-desktop:~$ clang++ -xc++ -S -o test.S - <<< 'extern "C" void f(); void f() {}' && cat test.S
        .text
        .file   "-"
        .globl  f                               # -- Begin function f
        .p2align        4, 0x90
        .type   f,@function
f:                                      # @f
        .cfi_startproc
# %bb.0:
        pushq   %rbp
        .cfi_def_cfa_offset 16
        .cfi_offset %rbp, -16
        movq    %rsp, %rbp
        .cfi_def_cfa_register %rbp
        popq    %rbp
        .cfi_def_cfa %rsp, 8
        retq
.Lfunc_end0:
        .size   f, .Lfunc_end0-f
        .cfi_endproc
                                        # -- End function
        .ident  "clang version 11.0.0 (https://github.com/llvm/llvm-project.git f5df584a5077e726ad851ccfe8496deda3e5ef07)"
        .section        ".note.GNU-stack","",@progbits
        .addrsig
```

```
milovidov@milovidov-desktop:~$ clang++ -xc++ -S -o test.S - <<< 'void f(); void f() {}' && cat test.S
        .text
        .file   "-"
        .globl  _Z1fv                           # -- Begin function _Z1fv
        .p2align        4, 0x90
        .type   _Z1fv,@function
_Z1fv:                                  # @_Z1fv
        .cfi_startproc
# %bb.0:
        pushq   %rbp
        .cfi_def_cfa_offset 16
        .cfi_offset %rbp, -16
        movq    %rsp, %rbp
        .cfi_def_cfa_register %rbp
        popq    %rbp
        .cfi_def_cfa %rsp, 8
        retq
.Lfunc_end0:
        .size   _Z1fv, .Lfunc_end0-_Z1fv
        .cfi_endproc
                                        # -- End function
        .ident  "clang version 11.0.0 (https://github.com/llvm/llvm-project.git f5df584a5077e726ad851ccfe8496deda3e5ef07)"
        .section        ".note.GNU-stack","",@progbits
        .addrsig
```

Reply:
It turns out that CGO can not parse `exter "C" void f();` syntax.

I had to use `extern "C" {}` sections instead.